### PR TITLE
feat(python mongo join): Join Step [TCTC-1993]

### PR DESCRIPTION
**Pull request by raphaelvignes**
This PR aims to replicate mongo translator's join step logic from frontend to backend.

### CODE REVIEW

File: server/src/weaverbird/backends/mongo_translator/steps/join.py [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+
+from weaverbird.backends.mongo_translator.utils import column_to_user_variable
+from weaverbird.pipeline import Pipeline, steps
+from weaverbird.pipeline.steps import DomainStep, JoinStep
+
+logger = logging.getLogger(__name__)
+
+
+def translate_join(step: JoinStep) -> List:
+    mongo_pipeline: List[Any] = []
```
Comment (jGundermann) [inline server/src/weaverbird/backends/mongo_translator/steps/join.py:12]: could it be a list of dictionary ?
```suggestion
    mongo_pipeline: List[Dict] = []
```

**Review by jGundermann — COMMENTED** [review]
Some suggestions

File: server/src/weaverbird/backends/mongo_translator/steps/join.py (hunk 1 above)
Comment (raphaelvignes) [reply server/src/weaverbird/backends/mongo_translator/steps/join.py]: :+1:

**Review by jGundermann — APPROVED** [review]
LGTM

**Pull request merged**